### PR TITLE
Feat/api extension

### LINKED ISSUE #71: API extension
Currently the API has only one JSON endpoint: `/_api/hotspots/{locationId}.json`.

For allowing other apps to use the abstracted (and cached) API, it needs to be extended.
Suggestion:
* [x] `/_api/locations/`  --> Return all the available cities/locations with Name, Id and Position
* [x] `/_api/locations/{locationId}/` --> Returns pois, categories and languages for the location
* [x] `/_api/locations/{locationId}/pois/` -->  Return the pois for one location
* [x] `/_api/locations/{locationId}/categories/` -->  Return the categories for one location
* [x] `/_api/locations/{locationId}/languages/` -->  Return the languages for one location
* [x] `/_api/hotspots/{locationId}.json` --> Return the pois for one location (implemented already). Shall redirect to the endpoint above soon and then be deprecated.

### PR DESCRIPTION
**Pull request by ro-ka**
Closes #71

### CODE REVIEW

**Comment by aemkei** [comment]
Is there a way to push the Go / API changes here only and move the JS renaming to another PR? 
Otherwise it would be hard to merge the other branches from the Hackathon.

**Comment by aemkei** [comment]
LGTM
@mrMetalWood or @Poltergeist can you please review the Go part, too?

File: src/lib/constants/constants.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package constants
 
 const (
 	SiteName                = "Refugee Maps"
-	CitySpreadsheetId       = "1pg-73mda1ZBtGZAFdkt2gh6XXCHxPuYnaWhNQbrcDx0"
+	LocationSpreadsheetId       = "1pg-73mda1ZBtGZAFdkt2gh6XXCHxPuYnaWhNQbrcDx0"
```
Comment (aemkei) [inline src/lib/constants/constants.go]: Strange formatting.

File: src/lib/locations/locations.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	"lib/position"
+	"lib/spreadsheet"
+	"lib/subdomain"
+)
+
+// Parse the location data
+func All(c appengine.Context) (cities []Location) {
+	sheetId := "0"
+	headerRow := 0
+	citiesData := spreadsheet.Get(c, constants.LocationSpreadsheetId, sheetId, headerRow)
```
Comment (aemkei) [inline src/lib/locations/locations.go]: Should be `locationsData`.

**Comment by aemkei** [comment]
:checkered_flag:

**Comment by ro-ka** [comment]
Updated.

**Pull request merged**